### PR TITLE
New Source Added. SQS S3 Notifications  and  SQS Messages

### PR DESCRIPTION
**Pull request by juan-coralogix**
# Description
This will add sqs support for s3 notifications (S3, Cloudtrail, Csv, VpcFlow, etc ) that uses Sqs as transport. 
Also will add Sqs Messages support. Retrieves the message in SQS and send it to Coralogix as Logs.

# How Has This Been Tested?

Locally

# Checklist:
- [ ] I have updated the versions in the SemanticVersion in template.yaml
- [X] I have updated the CHANGELOG.md
- [ ] I have created necessary PR to Terraform Module Repository (https://github.com/coralogix/terraform-coralogix-aws) if needed
- [ ] This change does not affect any particular component (e.g. it's CI or docs change)

### CODE REVIEW

**Review by daidokoro — APPROVED** [review]
LGTM

File: src/lib.rs [hunk 1]
```diff
@@ -107,6 +107,33 @@ pub async fn function_handler(
             crate::process::cloudwatch_logs(cloudwatch_event_log, coralogix_exporter, config)
                 .await?;
         }
+        CombinedEvent::Sqs(sqs_event) => {
+            debug!("SQS Event: {:?}", sqs_event.records[0]);
+            if let Some(message) = &sqs_event.records[0].body {
+                if config.integration_type != IntegrationType::Sqs  {
+                    let records = serde_json::from_str::<serde_json::Value>(message)?;               
+                    debug!("SQS S3 EVENT Detected");
+                    let bucket = records["Records"][0]["s3"]["bucket"]["name"]
```
Comment (RafalSumislawski) [inline src/lib.rs:116]: SQS lambda trigger hash batching enabled by default (at least when created via CF / SAM). That means we should handle all records, not just the zeroth one (like we could reasonably do with other type of events)

File: template.yaml [hunk 2]
```diff
@@ -740,6 +778,66 @@ Resources:
...
+      Principal: s3.amazonaws.com
+      SourceAccount: !Ref AWS::AccountId
+      SourceArn: !Sub "arn:aws:s3:::${S3BucketName}"
+
+  LambdaFunctionNotificationSubscriptionWithSQSTopic:
+    Type: AWS::SQS::Subscription
+    Condition: UseSQSTopicARNWithNotification
+    Properties:
+      Protocol: email
+      Endpoint: !Ref NotificationEmail
```
Comment (RafalSumislawski) [inline template.yaml:839]: Why is there something e-mail related in this PR?

File: src/combined_event.rs [hunk 3]
```diff
@@ -28,6 +30,10 @@ impl<'de> Deserialize<'de> for CombinedEvent {
                 Ok(CombinedEvent::Sns(
                     SnsEvent::deserialize(raw_value).map_err(de::Error::custom)?,
                 ))
+            } else if records[0].get("body").is_some() {
```
Comment (RafalSumislawski) [inline src/combined_event.rs:33]: Checking if "eventSource" is"aws:sqs" would be a more reliable method of recognising SQS events.

File: template.yaml (hunk 2 above)
Comment (RafalSumislawski) [reply template.yaml:839]: Oh, I see, these are the notifications about a failure of the lambda.

**Review by RafalSumislawski — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (juan-coralogix) [reply src/lib.rs]: Added For

File: src/combined_event.rs (hunk 3 above)
Comment (juan-coralogix) [reply src/combined_event.rs]: solved.

**Pull request merged**